conversations.go

```diff
@@ -0,0 +1,175 @@
+package main
+
+import (
+	"accounts-service/auth"
+	"accounts-service/models"
+	conversationsv1 "accounts-service/protorepo/noted/accounts/v1"
+	"accounts-service/validators"
+	"context"
+
+	"go.uber.org/zap"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type conversationsAPI struct {
+	conversationsv1.UnimplementedConversationsAPIServer
+
+	auth   auth.Service
+	logger *zap.Logger
+	repo   models.ConversationsRepository
+}
+
+var _ conversationsv1.ConversationsAPIServer = &conversationsAPI{}
+
+func (server *conversationsAPI) CreateConversation(ctx context.Context, in *conversationsv1.CreateConversationRequest) (*conversationsv1.CreateConversationResponse, error) {
+	err := validators.ValidateCreateConversationRequest(in)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	groupConversations, err := server.repo.List(ctx, &models.AllConversationsFilter{ID: in.GroupId})
+
+	for _, conv := range groupConversations {
+		res := conv.Title == in.Title
+		if res {
+			return nil, status.Error(codes.Internal, "a conversations with this title always exist")
+		}
+	}
```

Comment:
I think this is unnecessary for now. Also this endpoint is called internally on group creation so this shouldn't be a worry.

Reply:
You right, i just want to take advance for the future. Is it bad or i can let it for now ?

Reply:
In my opinion should remove it because I don't think this implementation is optimal and besides we can always find it in the commit log if one day we decide to use it.

**By all means, if you want to keep it in the code, no problems, I'm just giving my opinion and yours count just as much**. I just think that we should remove it because this isn't the right way to check for a duplicate title. The "right" way would be to `List()` all the conversations with filter `{title: title}`.

Imagine if the group has 20,000 conversations, you just loaded 20,000 conversations in-memory in the accounts-service (of course this is exaggerated but you get the point). So it's better to let the DB do the "work" of filtering conversations.

But even with this approach it might not work, what if between the call to `List` and the call to `Insert` someone else calls the endpoint and tries to add a conversation with the same title. Well you might end up with two conversations with the same name in your database. So this code might allow two conversations with the same name to be created and allow for some very rare and unexpected bugs especially if we expect that all conversations name are unique. This is where transactions could come into play but it's not worth it.

So the **only** correct way to implement this feature is to add a **uniqueness constraint on the `conversation.title` field in MongoBD**. You need to create an index on this field and make it unique for the whole collection. But again, we might do this at a later date because for now only one conversation exists per group.

Besides, adding a uniqueness constraint through indexes is self-documenting, works everywhere, with every use of the DB possible, even outside this endpoint and people get a clear error message saying there should be no duplicate titles when trying to insert a new conversation for which the title is already taken.

Let me know what you end up doing 😊

Reply:
now that i read all i presume that it is going to be done in mongo db call.
I let it now but going to erase it in next pr